magefiles/utils.go

```diff
@@ -230,3 +231,60 @@ func cleanupQuay(quayService quay.QuayService, quayOrg string) error {
 	}
 	return nil
 }
+
+func cleanupQuayTags(quayService quay.QuayService, organization, repository string) error {
+	workerCount := 10
+	var wg sync.WaitGroup
+
+	var allTags []quay.Tag
+	errChan := make(chan error)
+
+	page := 1
+	for {
+		tags, hasAdditional, err := quayService.GetTagsFromPage(organization, repository, page)
+		if err != nil {
+			errChan <- fmt.Errorf("error getting tags of `%s` repository of `%s` organization on page `%d`, error: %s", repository, organization, page, err)
+			continue
+		}
+		allTags = append(allTags, tags...)
+		if !hasAdditional {
+			break
+		}
+		page++
+	}
+
+	wg.Add(workerCount)
+
+	var allTagsMutex sync.Mutex
+	for i := 0; i < workerCount; i++ {
+		go func(startIdx int, allTags []quay.Tag, allTagsMutex *sync.Mutex, errChan chan<- error, wg *sync.WaitGroup) {
+			defer wg.Done()
+			for idx := startIdx; idx < len(allTags); idx += workerCount {
+				allTagsMutex.Lock()
+				tag := allTags[idx]
+				allTagsMutex.Unlock()
+				if time.Unix(tag.StartTS, 0).Before(time.Now().AddDate(0, 0, -7)) {
+					deleted, err := quayService.DeleteTag(organization, repository, tag.Name)
+					if err != nil {
+						errChan <- fmt.Errorf("error during deletion of tag `%s` in repository `%s` of organization `%s`, error: `%s`\n", tag.Name, repository, organization, err)
```

Comment:
same question for this line

Reply:
hmm doesn't need the value from `errChan` (sent to the channel here) be received first before the loop in this routine can continue?